### DISCUSSION TITLE
The message that is resolve with key XXXXX is not supported for jit compilation

### DISCUSSION THREAD

**Discussion by DidoMarchet** [intlify/vue-i18n#1516] [General]
Hi guys!
In Nuxt 3 i18n (@nuxtjs/i18n": "^8.0.0-rc.2") I've got this message:
`[intlify] the message that is resolve with key 'navigation.collections') is not supported for jit compilation`
Does someone know how to fix and explain what means? 

Because I didn't find anything online.

Really thanks and kind regards,

**Comment by DidoMarchet** [comment]
Hi,
does someone have any ideas?

thanks and kin regards,
Davide

**Comment by dev-caballero** [comment]
Hi, 

I was presenting this issue when importing i18n.config.ts file the wrong way in Nuxt 3.

I forgot to include the return in the callback function.

```ts
export default defineI18nConfig(() => {
  return {
    legacy: false,
    locale: "pt",
    messages: {
      es,
      en,
      pt,
    },
  };
});
```

**Comment by GaborTorma** [comment]
I had the same error message when I use defineI18nLocale function or import another large js/ts file into locale file.
You can test it in this [minimal reproduction repo](https://stackblitz.com/edit/nuxt-starter-zi8anv?file=locales%2Fen-warn.ts).

**Comment by GaborTorma** [comment]
Solved in @nuxtjs/i18n edge release: https://github.com/nuxt-modules/i18n/issues/2394#issuecomment-1712751292

**Reply by DidoMarchet** [reply]
Hi, thanks for the update!
Kind regards,
Davide
